massa-models/src/operation.rs

```diff
@@ -412,7 +414,18 @@ impl std::fmt::Display for Operation {
 /// signed operation
 pub type SecureShareOperation = SecureShare<Operation, OperationId>;
 
-impl SecureShareContent for Operation {}
+impl SecureShareContent for Operation {
+    fn compute_hash(&self, content_serialized: &[u8], content_creator_pub_key: &PublicKey) -> Hash {
+        let mut hash_data = Vec::new();
+        // Note: Add chain id before content hash in order to avoid replay attacks,
+        //       otherwise someone can copy an operation from testnet and execute it on main net
+        // Note 2: This makes the OperationId unique (per chain id)
```

Comment:
If I understand correctly, with how this is implemented, we cannot know what ChainId an operation is from. Would we ever want that? (I don't think so, but want to make sure).

Reply:
Only way is for the signer but otherwise no. @damip any comments on this point?

Reply:
The use case [like in ethereum ](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) is only to verify that the tx is in its expected network for safety. Otherwise we would have added a field "chain_id" in the Operation structure itself but that would have increased the operation size, network load and reduces throughput